scripts/Calibration/tofpd/diagnostics.py

```diff
@@ -223,3 +263,275 @@ def difc_plot2d(calib_new, calib_old=None, instr_ws=None, mask=None, vrange=(0,1
     fig.show()
 
     return fig, ax
+
+
+def __calculate_strain(obs, exp: np.ndarray):
+    return np.asarray(list(obs.values())[1:-2]) / exp
+
+
+def __calculate_difference(obs, exp: np.ndarray):
+    obs_ndarray = np.asarray(list(obs.values())[1:-2])
+    return np.abs(obs_ndarray - exp) / exp
+
+
+def __calculate_dspacing(obs):
+    return np.asarray(list(obs.values())[1:-2])
+
+
+def __create_outputws(donor, numSpec, numPeaks):
+    '''The resulting workspace needs to be added to the ADS'''
+    # convert the d-space table to a Workspace2d
+    if donor:
+        donor = mtd[str(donor)]
+    else:
+        donor = 'Workspace2D'
+    output = WorkspaceFactory.create(donor, NVectors=numSpec,
+                                     XLength=numPeaks, YLength=numPeaks)
+    output.getAxis(0).setUnit('dSpacing')
+    return output
+
+
+def collect_peaks(wksp, outputname: str, donor=None, infotype: str = 'strain'):
+    if infotype not in ['strain', 'difference', 'dspacing']:
```

Comment:
This is a public function, so I'm going to drone on with my requests for  :smiley: formal doctrings

Reply:
Also, you could add typehints for `wksp` since I see you're adding hints for the other arguments. I think is something like `Union[str, PeaksWorkspace]`. You'll have to import `PeaksWorkspace` from `mantid.api` or `mantid.dataobjects`, I believe.